client/aws/errors.go

```diff
@@ -11,8 +11,23 @@ const (
 	ELBConfigurationMismatch = "already exists and it is configured with different parameters"
 )
 
-var (
-	malformedAmazonAccountIDError   = errgo.New("amazon account ID can only contain numbers")
-	wrongAmazonAccountIDLengthError = errgo.New("amazon account ID has the wrong size")
-	emptyAmazonAccountIDError       = errgo.New("amazon account ID cannot be empty")
-)
+var malformedAmazonAccountIDError = errgo.New("amazon account ID can only contain numbers")
+
+// IsMalformedAmazonAccountID asserts malformedAmazonAccountIDError.
+func IsMalformedAmazonAccountID(err error) bool {
+	return errgo.Cause(err) == malformedAmazonAccountIDError
+}
+
+var wrongAmazonAccountIDLengthError = errgo.New("amazon account ID has the wrong size")
+
+// IsWrongAmazonIDLength asserts wrongAmazonAccountIDLengthError.
+func IsWrongAmazonIDLength(err error) bool {
```

Comment:
When the error is `wrongAmazonAccountIDLengthError` the matcher should read `IsWrongAmazonAccountIDLength`.